test/siren-action-behavior.js

```diff
@@ -110,72 +110,76 @@ suite('siren-action-behavior', function() {
 			result.then(function() {
 				sinon.assert.calledOnce(window.fetch);
-				var request = window.fetch.getCall(0).args[0];
-				expect(request.url).to.equal('http://api.x.io/orders/42/items');
+				var request = window.fetch.getCall(0).args[0];
+				expect(request.url).to.equal('http://api.x.io/orders/42/items');
 				expect(request.method).to.equal('POST');
 			});
 			return result;
 		});
-	});
-
-	suite('enqueue action unit tests', function() {
-		var res1, res2;
-		var fetchStub;
-		setup(function() {
-			fetchStub = sinon.stub(window.d2lfetch, 'fetch');
-			res1 = new window.Response(JSON.stringify(testEntity), {
-				status: 200,
-				headers: {
-					'Content-type': 'application/json'
-				},
-			});
-			res2 = new window.Response(null, {
-				status: 204
-			});
-		});
-
-		teardown(function() {
-			fetchStub.restore();
-		});
-
-		test('enqueues actions', function() {
-			fetchStub.withArgs('http://api.x.io/orders/42/items').returns(
-				new Promise(function(resolve) {
-					setTimeout(function() {
-						resolve(res1);
-					});
-				})
-			);
-			fetchStub.withArgs('http://api.x.io/orders/42/items/5').returns(
-				new Promise(function(resolve) {
-					resolve(res2);
-				})
-			);
-			var firstCall = element.performSirenAction(testAction).then(function() {
-				expect(fetchStub.callCount).to.equal(1);
-			});
-			element.performSirenAction(testAction2);
-			return firstCall;
-		});
-
-		test('immediate actions skip queue', function() {
-			fetchStub.withArgs('http://api.x.io/orders/42/items').returns(
-				new Promise(function(resolve) {
-					setTimeout(function() {
-						resolve(res1);
-					});
-				})
-			);
-			fetchStub.withArgs('http://api.x.io/orders/42/items/5').returns(
-				new Promise(function(resolve) {
-					resolve(res2);
-				})
-			);
-			var firstCall = element.performSirenAction(testAction).then(function() {
-				expect(fetchStub.callCount).to.equal(2);
-			});
-			element.performSirenAction(testAction2, null, true);
-			return firstCall;
-		});
-	});
+	});
+
+	suite('enqueue action unit tests', function() {
+		var res1, res2;
+		var fetchStub;
+		setup(function() {
+			fetchStub = sinon.stub(window.d2lfetch, 'fetch');
+			res1 = new window.Response(JSON.stringify(testEntity), {
+				status: 200,
+				headers: {
+					'Content-type': 'application/json'
+				},
+			});
+			res2 = new window.Response(null, {
+				status: 204
+			});
+		});
+
+		teardown(function() {
+			fetchStub.restore();
+		});
+
+		test('enqueues actions', function() {
+			fetchStub.withArgs('http://api.x.io/orders/42/items').returns(
+				new Promise(function(resolve) {
+					setTimeout(function() {
+						resolve(res1);
+					});
+				})
+			);
+			fetchStub.withArgs('http://api.x.io/orders/42/items/5').returns(
+				new Promise(function(resolve) {
+					resolve(res2);
+				})
+			);
+			var firstCall = element.performSirenAction(testAction).then(function() {
+				setTimeout(function() {

```

Comment:
Why the `setTimeout`? Is the promise resolving too early?

I expect that the promise resolves only after the action has been performed.

Reply:
https://github.com/Brightspace/polymer-siren-behaviors/issues/45 
Will remove `setTimeout` and find another solution to make the test case pass.